### PR TITLE
 Add post-run hook to create bucket in minio 

### PR DESCRIPTION
**Pull request by jamesc**
This was currently done by the provision.sh but this means if you
fire up the minio service you end up with an unconfigured instance.

This puts the config into a post-run hook (and means a docker 
exported habitat/builder-minio now comes up properly configured)

Signed-off-by: James Casey <james@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by jamesc** [comment]
Once the package is published we'd need to do some cleanup/removals
* remove the configure-minio bits from .studiorc
* remove the bucket setup code from on-prem-builder

Doing it now seems to break the tests

**Comment by chefsalim** [comment]
Just need to make sure we understand if any changes will be needed for on-prem script as well..

**Comment by jamesc** [comment]
@chefsalim. I created habitat-sh/on-prem-builder#96 to remove the duplicate configuration from `provision.sh`.  As there is already logic in there to only create if it doesn't exist you can merge this PR before the other with no adverse effect

File: components/builder-minio/habitat/hooks/post-run [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+exec 2>&1
+
+export AWS_ACCESS_KEY_ID="{{cfg.key_id}}"
+export AWS_SECRET_ACCESS_KEY="{{cfg.secret_key}}"
+
+{{ #if cfg.use_ssl }}
+MINIO_ENDPOINT="--endpoint-url https://localhost:9000 --no-verify-ssl"
```
[the remark this answers is not in the document]
Comment (jamesc) [reply components/builder-minio/habitat/hooks/post-run]: I could, but no real need as I know it always binds on localhost as well as the public IP

**Pull request merged**